test/export/StructureDefinitionExporter.test.ts

```diff
@@ -343,6 +343,32 @@ describe('StructureDefinitionExporter', () => {
     expect(exported.derivation).toBe('constraint');
   });
 
+  it('should export sub-extensions, with similar starting names and different types', () => {
+    const ruleString = new OnlyRule('extension[FooFoo].value[x]');
+    ruleString.types = [{ type: 'string' }];
+    const ruleDecimal = new OnlyRule('extension[FooBar].value[x]');
+    ruleDecimal.types = [{ type: 'decimal' }];
+    const exParent = new Extension('Parent');
+
+    const FooFooCardRule = new CardRule('extension[FooFoo]');
+    FooFooCardRule.min = 1;
+    FooFooCardRule.max = '1'; // * extension[sliceB].extension 1..1
+
+    const FooBarCardRule = new CardRule('extension[FooBar]');
+    FooBarCardRule.min = 0;
+    FooBarCardRule.max = '1'; // * extension[sliceB].extension 0..0
+
+    const containsRule = new ContainsRule('extension');
+    containsRule.items = [{ name: 'FooFoo' }, { name: 'FooBar' }];
```

Comment:
Thanks for the test! I think I know why this test isn't catching the original bug. In the original report, the first extension name, `forward` is entirely contained within the second extension name, `forwardByteSize`. In this test, that is not the case, `FooFoo`  is not a prefix of `FooBar`. If you change this test so that the `FooFoo` extension is just named `Foo`, so it has a name that is a prefix of `FooBar`, then the test should correctly fail.